File: server/View/Island.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;

namespace Com.GitHub.GlowingPotato.IslandDecimation.Server.View {
    public class Island : IViewModel {
        public bool HasUpdated {
            get {
                return ScoreHasUpdated || ExperienceHasUpdated || Resources.HasUpdated || Terrain.HasUpdated || Buildings.HasUpdated || AngleHasUpdated;
            }
        }

#region Score
        double score;
        bool ScoreHasUpdated;

        [JsonProperty("score")]
        public double Score {
            get {
                return score;
            }
            set {
                score = value;
                ScoreHasUpdated = true;
            }
        }

        public bool ShouldSerializeScore() {
            bool val = ScoreHasUpdated;
            if (val) {
                ScoreHasUpdated = false;
            }
            return val;
        }
#endregion
#region Experience
        double experience;
        bool ExperienceHasUpdated;

        [JsonProperty("experience")]
        public double Experience {
            get {
                return experience;
            }
            set {
                experience = value;
                ExperienceHasUpdated = true;
            }
        }

        public bool ShouldSerializeExperience() {
            bool val = ExperienceHasUpdated;
            if (val) {
                ExperienceHasUpdated = false;
            }
            return val;
        }
#endregion
#region Resources
        [JsonProperty("resources")]
        public Resources Resources;

        public bool ShouldSerializeResources() {
            return Resources.HasUpdated;
        }
#endregion
#region Terrain
        [JsonProperty("terrain")]
        public Terrain Terrain;

        public bool ShouldSerializeTerrain() {
            return Terrain.HasUpdated;
        }
#endregion
#region Buildings
        [JsonProperty("buildings")]
        public ViewModelList<Building> Buildings;

        public bool ShouldSerializeBuildings() {
            return Buildings.HasUpdated;
        }
#endregion
#region Angle
        double angle;
        bool AngleHasUpdated;

        [JsonProperty("angle")]
        public double Angle {
            get {
                return angle;
            }
            set {
                angle = value;
                AngleHasUpdated = true;
            }
        }

        public bool ShouldSerializeAngle() {
            bool val = AngleHasUpdated;
            if (val) {
                AngleHasUpdated = false;
            }
            return val;
        }
#endregion
    }
}

File: client/src/main/java/com/github/glowingpotato/islanddecimation/view/Island.java
package com.github.glowingpotato.islanddecimation.view;

public class Island {
	
	private Double score;
	private Double experience;
	private Resource resources;
	private Terrain terrain;
	private Double angle;
	
	public Island(Double score, Double experience, Resource resources, Terrain terrain, Double angle) {
		this.score = score;
		this.experience = experience;
		this.resources = resources;
		this.terrain = terrain;
		this.angle = angle;
	}
	
	public void update(Island island) {
		if(island.getScore() != null)
			setScore(island.getScore());
		if(island.getExperience() != null)
			setExperience(island.getExperience());
		if(island.getResources() != null)
			setResources(island.getResources());
		if(island.getTerrain() != null)
			setTerrain(island.getTerrain());
		if(island.getAngle() != null)
			setAngle(island.getAngle());
	}

	public Double getScore() {
		return score;
	}

	public void setScore(Double score) {
		this.score = score;
	}

	public Double getExperience() {
		return experience;
	}

	public void setExperience(Double experience) {
		this.experience = experience;
	}

	public Resource getResources() {
		return resources;
	}

	public void setResources(Resource resources) {
		this.resources = resources;
	}

	public Terrain getTerrain() {
		return terrain;
	}

	public void setTerrain(Terrain terrain) {
		this.terrain = terrain;
	}

	public Double getAngle() {
		return angle;
	}

	public void setAngle(Double angle) {
		this.angle = angle;
	}
}
